x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -248,6 +262,48 @@ export class AgentlessConnectorsInfraService {
     }
   };
 
+  public getAgentPolicyForConnectorId = async ({
+    connectorId,
+  }: {
+    connectorId: string;
+  }): Promise<PackagePolicyMetadata | null> => {
+    const allPolicies = await this.getConnectorPackagePolicies();
+
+    const policies = getPoliciesByConnectorId(allPolicies, connectorId);
+
+    if (!policies || policies.length === 0) {
+      return null;
+    }
+
+    const [policy] = policies;
+
+    if (policy && policy.agent_policy_ids.length > 0) {
+      const policyId = policy!.agent_policy_ids[0];
+
+      const listAgentsResponse = await this.agentService.asInternalUser.listAgents({
+        kuery: `fleet-agents.policy_id:${policyId}`,
+        showInactive: false,
+      });
+
+      if (!listAgentsResponse || listAgentsResponse.agents.length === 0) {
+        // If no agents assigned to policy, just return the policy
+        return policy;
+      } else {
+        // Return the first (and only) agentless host associated with this policy
```

Comment:
Shall we check if there are several agents and warn if so?

Reply:
Addressed in https://github.com/elastic/kibana/pull/210222/commits/1897dacd79d7b8104a7a203ade9c09566ef0d4f8